homeassistant/components/matter/adapter.py

```diff
@@ -52,11 +52,27 @@ def register_platform_handler(
 
     async def setup_nodes(self) -> None:
         """Set up all existing nodes and subscribe to new nodes."""
+        initialized_nodes: set[int] = set()
         for node in self.matter_client.get_nodes():
+            if not node.available:
+                # ignore un-initialized nodes at startup
+                # catch them later when they become available.
+                continue
+            initialized_nodes.add(node.node_id)
             self._setup_node(node)
```

Comment:
So in the past, we called setup node for all of them, even when unavailable. This has mainly bad implications when the node aren't interviewed (or when the interview data is old/stale): They don't get the platform setup correctly.

However, what causes not calling setup node at all? Will the node still be shown? :thinking: 

I feel we should display even uninterviewed node. Basically have a "bare" Matter device. This would allow also delete nodes where interviewing failed/crashed or whatnot.

Reply:
We cant display uninterviewed nodes, core simply doesn't allow creating entity less devices.

Reply:
> We cant display uninterviewed nodes, core simply doesn't allow creating entity less devices.

Hm, I see :cry: 

I guess the next best thing would be to have a diagnostics entity, e.g. last interviewed or similar to show them.

> If the device was available once before (which it will be in 999,9% of the cases) you will see the node displayed with unavailable entities just fine

It is an edge case I agree! But I have seen it, and I have seen folks on Discord hitting it. E.g. a bad RF connection cause it.

Anyhow, this is definitly not for this PR! Thanks for the insights! :+1: